packages/mobile/src/app/selectors.ts

```diff
@@ -180,29 +182,48 @@ export const registrationStepsSelector = createSelector(
     biometryEnabledSelector,
     activeScreenSelector,
     recoveringFromStoreWipeSelector,
+    skipVerificationSelector,
   ],
-  (chooseRestoreAccount, biometryEnabled, activeScreen, recoveringFromStoreWipe) => {
+  (
+    chooseRestoreAccount,
+    biometryEnabled,
+    activeScreen,
+    recoveringFromStoreWipe,
+    skipVerification
+  ) => {
+    let steps
+    let totalSteps
+    let step
     if (recoveringFromStoreWipe) {
-      return {
-        step: storeWipeRecoverySteps[activeScreen as StoreWipeRecoveryScreens],
-        totalSteps: 3,
-      }
+      steps = storeWipeRecoverySteps
+      totalSteps = 3
+      step = storeWipeRecoverySteps[activeScreen as StoreWipeRecoveryScreens]
+    } else if (chooseRestoreAccount) {
+      steps = restoreAccountSteps
+      totalSteps = 5
+      step = restoreAccountSteps[activeScreen as RestoreAccountScreens]
+    } else {
+      steps = createAccountSteps
+      totalSteps = 4
+      step = createAccountSteps[activeScreen as CreateAccountScreens]
     }
 
-    if (chooseRestoreAccount) {
-      if (biometryEnabled) {
-        return { step: restoreAccountSteps[activeScreen as RestoreAccountScreens], totalSteps: 5 }
+    if (!biometryEnabled) {
+      if (Object.keys(steps).includes(Screens.EnableBiometry)) {
+        totalSteps--
+        step =
+          step > (steps as Record<Screens.EnableBiometry, number>)[Screens.EnableBiometry]
+            ? step - 1
+            : step
       }
-      // remove biometry screen from step
-      const step = restoreAccountSteps[activeScreen as RestoreAccountScreens]
-      return { step: step > 3 ? step - 1 : step, totalSteps: 4 }
+    }
+    if (skipVerification) {
```

Comment:
```suggestion
    if (skipVerification && totalSteps > 0) {
```
this is pretty dumb but feels a little safer